cernopendata/modules/records/serializers/schemas/schemaorg_schemas.py

```diff
@@ -0,0 +1,149 @@
+    id_ = fields.Method('get_identifier', dump_to='@id')
+
+    # Minimum functional
+    type_ = fields.Method('get_type', dump_to='@type')
+    name = fields.Str(attribute='title', dump_to='name')
+    description = fields.Method('get_description', dump_to='description')
+    identifier = fields.Method('get_identifier', dump_to='identifier')
+    url = fields.Method('get_url', dump_to='url')
+    creator = fields.Method('get_creator', dump_to='creator')
+    date_created = fields.Str(attribute='date_published',
+                              dump_to='dateCreated')
+    date_published = fields.Str(attribute='date_published',
+                                dump_to='datePublished')
+    publisher = fields.Method('get_publisher', dump_to='publisher')
+
+    # Minimum operational
+    # license = fields.Str('license', dump_to='license')
+    # data_standard = fields.Str('dataStandard', dump_to='dataStandard')
+    # date_modified = fields.Str('dateModified', dump_to='dateModified')
+    # structure = fields.Str('structure', dump_to='structure')
+    # access_url = fields.Str('accessUrl', dump_to='accessUrl')
+    # access_interface = fields.str('accessInterface',
+    #                              dump_to='accessInterface')
+
+    def get_creator(self, obj):
+        """Get creator(s) based on authors or collaboration field."""
+        authors = obj.get('authors', None)
+
+        if authors:
+            return [{"@type": "Person", 'name': x['name']} for x in authors]
+        else:
+            # Note: Can there be many Organizations in a Record?
+            #       Now only one is expected.
+            collaboration = obj.get('collaboration', {}).get('name', None)
+            if collaboration:
+                return {"@type": "Organization", 'name': collaboration}
+            return None
+
+    def get_description(self, obj):
+        """Get description based on abstract.description field."""
+        return obj.get('abstract', {}).get('description', None)
+
+    def get_identifier(self, obj):
+        """Get identifier based on doi or recid field."""
+        doi = obj.get('doi', None)
+
+        if doi:
+            return "https://doi.org/{}".format(doi)
+        else:
+            recid = obj.get('recid', None)
+            return "http://opendata.cern.ch/record/{}".format(recid)
+
+    def get_publisher(self, obj):
+        """Get publisher based on publisher field."""
+        # Organization or Person - The publisher of the creative work.
+        publisher = obj.get('publisher', None)
+        if publisher:
+            return {"@type": "Organization", 'name': publisher}
+        return None
+
+    def get_type(self, obj):
+        """Get resource type based on Record type."""
+        return "CreativeWork"
+
+    def get_url(self, obj):
+        """Get url based on recid field."""
+        recid = obj.get('recid', None)
+        return "http://opendata.cern.ch/record/{}".format(recid)
```

Comment:
We should probably not hard-code `opendata.cern.ch`, but use the base URL of the deployed instance here, so that if somebody is using the QA service, the link would stay pointing to QA?

Reply:
Introduced a `SITE_URL` config variable which is used. Didn't want to use `request.url_root` as it might give funny/incorrect values and it wouldn't be available if we do serializing outside request context.
Probably an application wide config variable wouldn't be a bad idea.